### PR TITLE
Make snippets explicit, so no syntax errors appear.

### PR DESCRIPTION
**Pull request by jim-portegies**
The goal is that provides snippets do not give syntax errors, and immediately show what is expected.

For instance, after executing the following lines:
```
Goal forall a : nat, (a > 0) -> True.
Proof.
Take a : nat.
```
the snippet suggestion is ```Assume that (0 = 0).```

### CODE REVIEW

File: shared/completions/tactics.json [hunk 1]
```diff
@@ -47,7 +47,7 @@
         "label": "Choose (*name_var*) := ((*expression*)).",
         "type": "type",
         "detail": "tactic",
-        "template": "Choose ${(*name_var*)} := (${(*expression*)}).",
+        "template": "Choose ${x} := (${ 0}).",
```
Comment (DikieDick) [inline shared/completions/tactics.json:50]: There is an extra space here.

**Review by DikieDick — COMMENTED** [review]
Looks good. 

My only remark is that now the tab functionality of jumping to the next 'hole' in the template is less intuitive (I don't know if this functionality is used a lot by students).
Previously the `(* comment *)` placeholders clearly indicated a position where the user could jump to.

**Review by DikieDick — DISMISSED** [review]

**Comment by DikieDick** [comment]
Unrelated to the changes proposed in this PR, but the template for the 'chain of inequalities' inserts an unnecessary closing bracket `)`: [./shared/completions/tactics.json#L77](https://github.dev/impermeable/waterproof-vscode/blob/feat-parsing-templates/shared/completions/tactics.json#L77).

File: shared/completions/tactics.json (hunk 1 above)
Comment (DikieDick) [reply shared/completions/tactics.json:50]: Ah, without the space the `0` character gets interpreted as the order this placeholder should be completed in ([see documentation](https://codemirror.net/docs/ref/#autocomplete.snippet)). The following template works as intented: `"Choose ${0:x} := (${1:0})."`.

Comment (jim-portegies) [reply shared/completions/tactics.json:50]: Thanks! That's a very good thing to know.

**Comment by DikieDick** [comment]
@jim-portegies I was mistaken in saying that the closing bracket is unnecessary, somehow I completely overlooked the opening bracket on the same line :)

My original reason for this comment is the following example:
1. Insert `We conclude that (0 = 0).` using the snippet completions.
2. Cursor moves into the `0 = 0` placeholder.
3. Suppose we want to insert a chain of inequalities. We continue with typing `&`, the snippet appears and we insert it into the document.
4. The final result is `We conclude that (& 3 < 5 = 2 + 3 ≤ 7 )).`, which produces a syntax error.

Note: This is works fine using the tactics panel.

**Comment by jim-portegies** [comment]
@DikieDick your point about the snippet insertion was very good. I tried to implement the changes

**Review by DikieDick — APPROVED** [review]
Looks good. 

The chain of inequalities works really nicely :)

**Pull request merged**